xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp

```diff
@@ -171,8 +171,10 @@ bool CEGLNativeTypeAmlogic::ProbeResolutions(std::vector<RESOLUTION_INFO> &resol
   RESOLUTION_INFO res;
   for (std::vector<std::string>::const_iterator i = probe_str.begin(); i != probe_str.end(); ++i)
   {
-    if(aml_mode_to_resolution(i->c_str(), &res))
-      resolutions.push_back(res);
+    if (((StringUtils::StartsWith(i->c_str(), "4k2k")) && (aml_support_h264_4k2k() > AML_NO_H264_4K2K)) || !(StringUtils::StartsWith(i->c_str(), "4k2k"))) {
```

Comment:
aml_support_h264_4k2k() from AMLUtils.cpp is checking video codec profile (basically parsing /sys/class/amstream/vcodec_profile) if h264 4K decoding is supported with codec or not. S805 cannot display or decode 4K so value of aml_support_h264_4k2k() is AML_NO_H264_4K2K (see https://github.com/xbmc/xbmc/blob/master/xbmc/utils/AMLUtils.h#L44) For 9xx devices there is no need for these checks because they all can display 4K and that's why I did not include 2160 resolution checks (S9xx have different resolution strings. Eg. M8 is 4k2k30hz while on 9xx it's 2160p60hz.

About brackets, will do it in cpl mins.